Devise

joredudiaz opened the pull request:
# Summary
- Devise installation
- Create an user with specific data (first_name, last_name)
# Trello Card
https://trello.com/c/5bPFFDPG

joredudiaz committed:
UPDATE
Gemfile/Gemfile.lock
devise install

config/environments/development.rb
devise email config

config/routes.rb
devise user routes

CREATE
app/models/user.rb

config/initializers/devise.rb
devise config

config/locales/devise.en.yml
devise i18n

db/migrate/20181025132807_devise_create_users.rb
users' schema

db/schema.rb

joredudiaz committed:
UPDATE app/models/user.rb rubocop fix, to new syntax when validate password confirmation

leoncalermo commented on app/models/user.rb:
```diff
@@ -0,0 +1,8 @@
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  validates :password, confirmation: true
```
i think that you forgot to validate the email and password  presence in the model 🤔  because with this validations, you could create a user just using `User.create()` and it will create a user with all the fields empty and we don't want that 😛

joredudiaz replied:
The email, password, first_name and last_name are validated; the email string is now validated as email.

joredudiaz committed:
UPDATE
app/models/user.rb
user fields validations

CREATE
app/validators/email_validator.rb
email validator

joredudiaz committed:
UPDATE app/validators/email_validator.rb fix syntax

leoncalermo commented on app/models/user.rb:
```diff
@@ -0,0 +1,10 @@
+
+  validates :email, presence: true, uniqueness: true, email: true
+  validates :password, confirmation: true, length: { in: 6..20 }
+  validates :first_name, :last_name, presence: true
```
you can put email first_name and last_name  in the same validation of presence, 
something like this 
`validates :email, :first_name, :last_name, presence: true`and you are missing password too 👀  hahaha.

joredudiaz replied:
I put the same validation for different fields at one line and I don't forget password this time. 😭 😄

joredudiaz committed:
UPDATE same validations for different fields in one line
